File: LP3/Lista1/Q3.java
import java.util.Random;
import java.util.Scanner;


class Main {
  public static void main(String[] args) {
  Scanner sc = new Scanner(System.in);

  int jogada1, jogada2, resultado, ponto=0, i=1;

  Random random = new Random();

  System.out.println("Bem vindo ao Craps");
  System.out.println("Partida: ");

  jogada1 = random.nextInt(6)+1;
  jogada2 = random.nextInt(6)+1;  

  System.out.println("Dado 1: "+jogada1+" Dado 2: "+jogada2);

  resultado = jogada1 + jogada2;

  System.out.println("Pontuação inicial: "+resultado);

  if(resultado == 7 || resultado==11) {
    System.out.println("Parabens voce venceu!");
    return;
  }
  else if(resultado==2 || resultado==3 || resultado==12) {
    System.out.println("Que pena, voce perdeu! Craps!");
    return;
  }

  while(resultado != ponto && ponto != 7){
    
    System.out.println("Rodada "+i);
    i++;
    
    jogada1 = random.nextInt(6)+1;
    jogada2 = random.nextInt(6)+1;  
  
    System.out.println("Dado 1: "+jogada1+" Dado 2: "+jogada2);
  
    ponto = jogada1 + jogada2;
    System.out.println("Pontuação da rodada: "+ponto);
    System.out.println("Pontuação desejada: "+resultado);
  }

  if(ponto==resultado) System.out.println("Parabens voce venceu!");
  else System.out.println("Que pena, voce perdeu! Craps!");

  sc.close();
    
  }
  
}

File: LP3/Atividades/Lista 2/Q3.py
import random


def main():

  i, ponto = 1, 0
  print("Bem vindo ao Craps")
  print("Partida: ")

  jogada1 = random.randint(1, 6)
  jogada2 = random.randint(1, 6)

  print("Dado 1: " + str(jogada1) + " Dado 2: " + str(jogada2))
  resultado = jogada1 + jogada2
  print("Pontuacao inicial: ", resultado)

  if (resultado == 7 or resultado == 11):
    print("Parabens voce venceu!")
    return
  elif (resultado == 2 or resultado == 3 or resultado == 12):
    print("Que pena, voce perdeu! Craps!")
    return

  while (resultado != ponto and ponto != 7):

    print("Rodada ", i)
    i += 1

    jogada1 = random.randint(1, 6)
    jogada2 = random.randint(1, 6)

    ponto = jogada1 + jogada2

    print("Dado 1: " + str(jogada1) + " Dado 2: " + str(jogada2))
    print("Pontuação da rodada: ", ponto)
    print("Pontuação desejada: ", resultado)

  if (ponto == resultado): print("Parabens, voce venceu")
  else: print("Que pena, voce perdeu! Craps!!")

  return


main()